src/server/storage/users.js

```diff
@@ -33,6 +34,12 @@
             delete this.password;
         }
 
+        // updates based on mongoid vs username
+        update(user) {
+            return this._db.updateOne({_id: this._id}, { $set: {username: this.username, email: this.email} })
```

Comment:
I don't really like the idea of changing usernames of existing users... emails are no big deal to be able to change but usernames are more work than they are worth

Reply:
the goal is to only let them change usernames if it's a new user

Reply:
For simplicity's sake, it would probably be better to require them to delete the user and create a new one. Otherwise, we will have some duplication of logic for validating the username for a pretty minimal benefit, imo (esp if they can only change their username if they have no projects). I am not sure if the convenience of editing the username in a single request is worth it - maybe if we get a request for changing usernames from outside users...

Thoughts?

Reply:
by moving that logic into a middleware I tried to avoid duplicating that logic. 
I could totally disallow editing of usernames but the endpoint is also allowing them to change the user email which IMO is a  useful feature and wouldn't be as problematic as changing username.

Reply:
> the goal is to only let them change usernames only if it's a new user

plus:
- they are in a group (usercreated to be in a group)
- the group owner is requesting the change

Reply:
changing usernames is not there anymore